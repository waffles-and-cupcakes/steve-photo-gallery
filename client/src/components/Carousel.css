.slick-slide .image {
  padding: 10px;
  text-align: center;
}
.slick-slide img {
  display: block;
  margin: auto;
}
.slick-slide img.slick-loading {
  border: 0;
}
.slick-slider {
  margin: 30px auto 50px;
}
.slick-arrow {
  display: inline-block;
  height: 100%;
}

.photoDescription {
  color: white;
  list-style: none;
  text-align: left;
  padding: 0;
  font-family: 'Montserrat', sans-serif;
  font-size: 0.8em;
  
}

.verified {
  margin-top: 5px;
  font-style: italic;
  color: #888;
}