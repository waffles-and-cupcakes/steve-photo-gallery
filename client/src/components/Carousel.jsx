import React from 'react';
import CarouselImage from './CarouselImage';
import CarouselThumbnail from './CarouselThumbnail';
import style from './Carousel.css';

class Carousel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      main: null,
      thumbnail: null,
      currentImageIndex: 0,
    };

    this.photosCount = this.props.photos.length;

    this.setMain = this.setMain.bind(this);
    this.setThumbnail = this.setThumbnail.bind(this);
    this.setCurrentImageIndex = this.setCurrentImageIndex.bind(this);
  }

  componentDidMount() {
    this.setState({
      main: this.sliderMain,
      thumbnail: this.sliderThumbnail,
    });
  }

  setMain(slider) {
    this.sliderMain = slider;
  }

  setThumbnail(slider) {
    this.sliderThumbnail = slider;
  }

  setCurrentImageIndex(index) {
    this.setState({currentImageIndex: index});
  }

  render() {
    return (
      <div className={style.container}>
        <CarouselImage
          setMain={this.setMain}
          sliderMain={this.sliderMain}
          photos={this.props.photos}
          thumbnail={this.state.thumbnail}
          setCurrentImageIndex={this.setCurrentImageIndex} />
        <ul className={style.photoDescription}>
          <li>{`${this.state.currentImageIndex + 1}/${this.photosCount}: ${this.props.photos[this.state.currentImageIndex].title}`}</li>
          {this.props.photos[this.state.currentImageIndex].isVerified ? <li className={style.verified}>Verified Photo</li> : ''}
        </ul>
        <CarouselThumbnail setThumbnail={this.setThumbnail} sliderThumbnail={this.sliderThumbnail} photos={this.props.photos} main={this.state.main} />
      </div>
    );
  }
}

export default Carousel;