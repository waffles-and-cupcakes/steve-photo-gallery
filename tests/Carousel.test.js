import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Carousel from './Carousel.jsx';

configure({adapter: new Adapter() });

const photos = [
  {
    title: 'A photo',
    isVerified: false,
    photoUrl: 'www.example.com/photo',
    thumbnailUrl: 'www.example.com/thumbnail',
    listingId: 1,
  },
  {
    title: 'Another photo',
    isVerified: true,
    photoUrl: 'www.example.com/photo',
    thumbnailUrl: 'www.example.com/thumbnail',
    listingId: 1,
  }
];

test('Carousel should render correctly', () => {
  const output = shallow(<Carousel photos={photos} />);
  expect(shallowToJson(output)).toMatchSnapshot();
});

test('Carousel should keep track of the current image index', () => {
  const output = shallow(<Carousel photos={photos} />);
  output.instance().setCurrentImageIndex(1);
  expect(output.state().currentImageIndex).toBe(1);
});