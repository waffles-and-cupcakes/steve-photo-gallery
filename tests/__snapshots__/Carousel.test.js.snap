// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correctly 1`] = `
<div>
  <CarouselImage
    photos={
      Array [
        Object {
          "isVerified": false,
          "listingId": 1,
          "photoUrl": "www.example.com/photo",
          "thumbnailUrl": "www.example.com/thumbnail",
          "title": "A photo",
        },
        Object {
          "isVerified": true,
          "listingId": 1,
          "photoUrl": "www.example.com/photo",
          "thumbnailUrl": "www.example.com/thumbnail",
          "title": "Another photo",
        },
      ]
    }
    setCurrentImageIndex={[Function]}
    setMain={[Function]}
  />
  <ul>
    <li>
      1/2: A photo
    </li>
  </ul>
  <CarouselThumbnail
    photos={
      Array [
        Object {
          "isVerified": false,
          "listingId": 1,
          "photoUrl": "www.example.com/photo",
          "thumbnailUrl": "www.example.com/thumbnail",
          "title": "A photo",
        },
        Object {
          "isVerified": true,
          "listingId": 1,
          "photoUrl": "www.example.com/photo",
          "thumbnailUrl": "www.example.com/thumbnail",
          "title": "Another photo",
        },
      ]
    }
    setThumbnail={[Function]}
    windowWidth={1024}
  />
</div>
`;
